### PR TITLE
Docker image for kotlin 

### PR DESCRIPTION
**Pull request by m-x-k**

### CODE REVIEW

File: kotlin/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM iron/java
+
+ENV VERSION 1.0.5
+
+ENV KOTLIN_HOME=/usr/share/kotlin
+ENV KOTLIN_COMPILER_URL https://github.com/JetBrains/kotlin/releases/download/v${VERSION}/kotlin-compiler-${VERSION}.zip
```
Comment (nikhilm) [inline kotlin/Dockerfile]: Please put all 3 env vars on the same `ENV` line to reduce layers. You can use the `=` form to achieve this.

File: kotlin/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM iron/java
```
Comment (nikhilm) [inline kotlin/Dockerfile]: Please use a tagged 1.7.0 or 1.8.0 version.

File: kotlin/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+ENV VERSION 1.0.5
+
+ENV KOTLIN_HOME=/usr/share/kotlin
+ENV KOTLIN_COMPILER_URL https://github.com/JetBrains/kotlin/releases/download/v${VERSION}/kotlin-compiler-${VERSION}.zip
+
+RUN apk update
+RUN apk add bash
```
Comment (nikhilm) [inline kotlin/Dockerfile]: `RUN apk --no-cache add bash` should do the same thing and save some space.

File: kotlin/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+Push:
+
+```sh
+docker push iron/kotlin
+```
+
+Updating Kotlin version:
+
+Go to https://github.com/JetBrains/kotlin/releases/ for the latest releases.
```
Comment (nikhilm) [inline kotlin/README.md]: Could you add instructions on how to use this image to build/run Kotlin applications?

**Comment by nikhilm** [comment]
Thanks for the PR @m-x-k . We would appreciate some clarifications and changes before merging it.

File: kotlin/Dockerfile (hunk 3 above)
Comment (m-x-k) [inline kotlin/Dockerfile]: Great. I didn't want to use 'apk upgrade'. It resulted in a JVM initialization error: 

> Invalid layout of java.lang.Thread at name

Something to watch out for.

File: kotlin/Dockerfile (hunk 1 above)
Comment (m-x-k) [inline kotlin/Dockerfile]: Thanks. I didn't realise multiple layers would be impacted by this. The KOTLIN_HOME wasn't used but VERSION is required for the other line so I've combined them into one.

**Comment by m-x-k** [comment]
You're right about the use of the JRE iron/java to run the source code. Image mainly required for: https://github.com/iron-io/dockerworker/pull/39

**Comment by treeder** [comment]
@m-x-k I think the main point of @nikhilm 's comments are to break this into a build image tagged with `dev` suffixes and a runtime image for running in production. This is to keep the runtime/production images small. 

Take the java-1.8 images for example. There is `iron/java:1.8-dev` and `iron/java:1.8`.  You can see here we're using the full JDK: https://github.com/iron-io/dockers/blob/master/java/java-1.8/dev/Dockerfile for the `dev` images. 

For packaging the compiled binary, you can see here we're only using the JRE, not the full JDK: https://github.com/iron-io/dockers/blob/master/java/java-1.8/Dockerfile

For kotlin, it looks like what you have created is a good dev/build image, but the runtime image should probably just include the `kotlin-runtime.jar` from the zip, and nothing else. 

Does that make sense?

**Comment by m-x-k** [comment]
@treeder @nikhilm thanks for the clarification. Looks like the kotlin executable script references kotlinc so its not possible to separate these as you would with the JRE. Perhaps in future builds this will be possible. In the meantime I have provided examples showing how to build with _iron/kotlin:dev_ and execute with _iron/java:1.8_. Hope this helps to clarify.

**Comment by nikhilm** [comment]
Thanks! This looks good. Will push a dev image soon.

**Comment by nikhilm** [comment]
@treeder could you do this? I don't have permissions.

**Comment by treeder** [comment]
Thanks @m-x-k !

**Comment by treeder** [comment]
Pushed new images, please try them.

**Pull request merged**